records.py

```diff
@@ -165,6 +165,11 @@ def dataset(self):
         # Create a new Tablib Dataset.
         data = tablib.Dataset()
 
+        # If the RecordCollection is empty, just return the empty set
+        # Check number of rows by typecasting to list
+        if len(list(self)) == 0:
```

Comment:
I got the context wrong, sorry - of course you don't want to raise an error. I somehow thoughts it's in a `first` method .. Sorry for the confusion.